WooCommerce/UITestsFoundation/Screens/TabNavComponent.swift

```diff
@@ -48,11 +48,12 @@ public final class TabNavComponent: ScreenObject {
     @discardableResult
     public func gotoOrdersScreen() throws -> OrdersScreen {
         // Avoid transitioning if it is already on screen
-        if !OrdersScreen.isVisible {
+        guard let orderScreen = try? OrdersScreen(), orderScreen.isLoaded else {
```

Comment:
Seeing this line and trying to understand how it works was quite mind-blowing. I mean the usage of comma, and referencing `ordersScreen in the right part.

I tried to rewrite this as nested statements (but with a rule to still keep referencing the `orderScreen` constant in the nested statement) and failed because it was out of scope. So far, my understanding is that it's impossible to represent it by two nested statements without instantiating orders screen once again.

I went from 'Oh, I finally understood it', to 'But still, how can this work...' several times, and still somewhere in the middle. I'd say that my misunderstanding of this is on the intuitive level 😅 🥲.

I think [this](https://medium.com/swlh/optional-binding-in-swift-ed15335372e7) post describes what's going on here most precisely.

So far, I understand this line as: we try to assign the Orders screen to `ordersScreen` constant, but only if said screen is loaded.

Also, I was focused on `introspection` word in PR name - and from what I could find, this is another name for reflection in Swift. But this is not what's happening here?

cc @tiagomar and @jostnes: ^ this might be interesting for you too.

Reply:
I probably used a too colorful language in the PR title. To me, at least in the context of that title, introspection meant `OrdersScreen` looking into itself (via a method it exposes) to verify its visibility. A dangerous word choice because indeed introspection has a lot of deeper and more complex meaning in various programming language 😳 Sorry for causing a tangent.

Regarding the `,` and how it works, you are on the right track with "we try to assign the Orders screen to `ordersScreen` constant, but only if said screen is loaded". The order is the other way around, though: we try to instantiate `OrderScreen()` into a constant called `orderScreen`, and if that works, we check if `.isLoaded` equals `true`.

I can see how that code might be a bit confusing if you had never seen the `,` before. I think the `guard` and its "inverse if" behavior adds to the complexity.

If we rewrite this as code using `if`, it would look like this:

**With `guard` and `,`**

```swift
guard let orderScreen = try? OrdersScreen(), orderScreen.isLoaded else {
    ordersTabButton.tap()
    return try OrdersScreen()
}

return orderScreen
```

**With `if` and `,`**

```swift
if let orderScreen = try? OrderScreen(), orderScreen.isLoaded {
    return orderScreen
} else {
    ordersTabButton.tap()
    return try OrdersScreen()
}
```

At this point, we can use a nested `if` because we have the conditional branch in which `orderScreen` is avialble

```swift
if let orderScreen = try? OrderScreen() {
    if orderScreen.isLoaded {
        return orderScreen
   } else {
        ordersTabButton.tap()
        return try OrdersScreen()
    }
} else {
    ordersTabButton.tap()
    return try OrdersScreen()
}
```

I tried looking in the [Swift documentation](https://docs.swift.org/swift-book/LanguageGuide/ControlFlow.html) for an official explanation of what `,` does, but couldn't find it. 🤔 

My mental model is that it's a super powered `&&` which allows you to access everything previously instantiated in the same condition. 

Hope this helps clarify?

Reply:
> To me, at least in the context of that title, introspection meant OrdersScreen looking into itself (via a method it exposes) to verify its visibility.

This is how I understood it too, in this context 🙂. But only because the internets were filled with reflection mentions, I wanted to make sure I'm not missing something subtle here.

> I can see how that code might be a bit confusing if you had never seen the `,` before. I think the guard and its "inverse if" behavior adds to the complexity.

Indeed, especially because I tried to rewrite it as two nested `guard` statements 😆 

> If we rewrite this as code using if, it would look like this:

Thank you for spending your time to provide me with examples - this is very helpful, and much appreciated by me 🙇